workflow/metrics/metrics.go

```diff
@@ -86,33 +88,44 @@ func (m Metrics) allMetrics() []prometheus.Metric {
 	return allMetrics
 }
 
-func (m Metrics) WorkflowAdded(phase v1alpha1.NodePhase) {
+func (m *Metrics) WorkflowAdded(key string, phase v1alpha1.NodePhase) {
+	if m.workflows[key] {
+		return
+	}
 	if _, ok := m.workflowsByPhase[phase]; ok {
+		m.workflows[key] = true
 		m.workflowsByPhase[phase].Inc()
 	}
 }
 
-func (m Metrics) WorkflowUpdated(fromPhase, toPhase v1alpha1.NodePhase) {
-	m.WorkflowDeleted(fromPhase)
-	m.WorkflowAdded(toPhase)
+func (m *Metrics) WorkflowUpdated(key string, fromPhase, toPhase v1alpha1.NodePhase) {
+	if fromPhase == toPhase || !m.workflows[key] {
+		return
+	}
+	m.WorkflowDeleted(key, fromPhase)
+	m.WorkflowAdded(key, toPhase)
 }
 
-func (m Metrics) WorkflowDeleted(phase v1alpha1.NodePhase) {
+func (m *Metrics) WorkflowDeleted(key string, phase v1alpha1.NodePhase) {
+	if !m.workflows[key] {
+		return
+	}
 	if _, ok := m.workflowsByPhase[phase]; ok {
+		delete(m.workflows, key)
```

Comment:
move one line higher?

Reply:
Yes sorry, this was lingering from when I used to have the check on the `if` statement